conan/tools/env/environment.py

```diff
@@ -332,7 +333,9 @@ def save_bat(self, file_location, generate_deactivate=True):
             result.append('set "{}={}"'.format(varname, value))
 
         content = "\n".join(result)
-        save(file_location, content)
+        # It is very important to save it correctly with utf-8, the Conan util save() is broken
```

Comment:
The ``save()`` is massively used, I think it might break in other place, high risk. I think what we should do:
- Try to improve it in 2.0 branch
- Add some special chars to our helper ``tmp_folder`` in develop2, see what breaks and what not